src/main/java/de/tum/in/www1/artemis/web/rest/plagiarism/PlagiarismCaseResource.java

```diff
@@ -147,6 +178,31 @@ public ResponseEntity<PlagiarismCaseInfoDTO> getPlagiarismCaseForExerciseForStud
         return ResponseEntity.ok(null);
     }
 
+    /**
+     * Retrieves plagiarismCases related to given exercise ids for the student for the exercises where the plagiarism comparison was confirmed and the student was notified
+     *
+     * @param courseId the id of the course
+     * @param exerciseIds the ids of the exercises
+     * @return a list of plagiarism case id and verdict values for the exercises only for the exercises where the comparison was confirmed and the student was notified
+     */
+    @GetMapping("courses/{courseId}/plagiarism-cases")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<Map<Long, PlagiarismCaseInfoDTO>> getPlagiarismCasesForExercisesForStudent(@PathVariable long courseId,
+            @RequestParam(name = "exerciseId") Set<Long> exerciseIds) {
+        log.debug("REST request to all plagiarism cases for student and exercises with ids: {}", exerciseIds);
+        Course course = courseRepository.findByIdElseThrow(courseId);
+        var user = userRepository.getUserWithGroupsAndAuthorities();
+        authenticationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, user);
+
+        List<PlagiarismCase> plagiarismCasePerExerciseList = plagiarismCaseRepository.findByStudentIdAndExerciseIdsWithPost(user.getId(), exerciseIds);
```

Comment:
I think this method does not check if the given exercise IDs belong to the course used for access check above.

Reply:
At the end of the day, we need to add some ways to always allow the all students who meet those two conditions below ([at here](https://github.com/ls1intum/Artemis/pull/5620/files/3671d7574d8d9c3a514af403d21939aaea4a6712#diff-30868ed17409051376cbb46d6aade4d5a85f41b6661b8b7a9e5d0483dd3e3f8cR61-R62)) to access these plagiarism cases to make their defence.
```
...
AND plagiarismCase.student.id = :userId
AND p.id IS NOT NULL   // p is Post
```

So I am not sure the complexity increase added by the course check gives us any benefit. (I mentioned complexity increase because we would need to check whether this is a course plagiarism case or exam plagiarism case and if so get the course from exam and so on). Instead, maybe I can just remove the course parameter from this request completely to not give a false sense of security. What do you think?

Reply:
Well, it does not look that much complicated as I initially expected, this seems to work locally, how does it look?
```java
@GetMapping("courses/{courseId}/plagiarism-cases")
@PreAuthorize("hasRole('USER')")
public ResponseEntity<Map<Long, PlagiarismCaseInfoDTO>> getPlagiarismCasesForExercisesForStudent(@PathVariable long courseId,
        @RequestParam(name = "exerciseId") Set<Long> exerciseIds) {
    log.debug("REST request to all plagiarism cases for student and exercises with ids: {}", exerciseIds);
    Course course = courseRepository.findByIdElseThrow(courseId);
    var user = userRepository.getUserWithGroupsAndAuthorities();
    authenticationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, user);

    List<PlagiarismCase> plagiarismCasePerExerciseList = plagiarismCaseRepository.findByStudentIdAndExerciseIdsWithPost(user.getId(), exerciseIds);
    for (PlagiarismCase plagiarismCase : plagiarismCasePerExerciseList) {
        var plagiarismCaseCourse = plagiarismCase.getExercise().getCourseViaExerciseGroupOrCourseMember();
        if (!plagiarismCaseCourse.getId().equals(courseId)) {
            throw new ConflictException("Plagiarism case with id " + plagiarismCase.getId() + " is not related to the given course id " + courseId, ENTITY_NAME, "courseMismatch");
        }
    }
    Map<Long, PlagiarismCaseInfoDTO> plagiarismCaseInfoDTOs = plagiarismCasePerExerciseList.stream()
            // the following line is already checked in the SQL statement, but we want to ensure it 100%
            .filter(plagiarismCase -> plagiarismCase.getPost() != null).collect(Collectors.toMap(plagiarismCase -> plagiarismCase.getExercise().getId(),
                    plagiarismCase -> new PlagiarismCaseInfoDTO(plagiarismCase.getId(), plagiarismCase.getVerdict())));

    return ResponseEntity.ok(plagiarismCaseInfoDTOs);
}
```

Reply:
Fixed as written above in https://github.com/ls1intum/Artemis/pull/5620/commits/ce3f41d456abb3db93f8eebebf6a8bc8d970f158